### DISCUSSION TITLE
Modify pore and throat dimensions

### DISCUSSION THREAD

**Discussion by elstavro** [PMEAL/OpenPNM#2549] [Q&A]
Hello, 
I'm recent to the openPNM and I'm for now working on single phase flow on a network imported and built from a 3D x-ray image.
I have managed to calculate the absolute permeability and hydraulic conductivity which is though quite higher than experimental evidence.

So I'm trying to manually impose a smaller `throat_diameter` (and/or throat length) to achieve a lower `hydraulic_size_factor` and eventually `hydraulic_conductivity` by e.g.: `geo['throat.diameter'] = 0.001*geo['throat.diameter']` but nothing changes.
Same when modifying `pore_diameter`.

However when I modify `throat.hydraulic_size_factors.throat`I get eventually a decreased permeability up to a point, and if I also reduce `throat.hydraulic_size_factors.pore1` and `throat.hydraulic_size_factors.pore2` I can achieve the 3 orders of magnitude of abs permeability I get from experiments.

I don't understand why decrease of diameter does not result in lower hydraulic size factor, since these parameter is used to calculate the hydraulic size factor (models/geometry/hydraulic_size_factors.py).
Do I need to update my network/geometry so that the hydraulic size factor of each modified throat or pore is recalculated?
If yes, how to do so?

I hope my question is clear enough... Thanks for the help!

**Comment by jgostick** [comment]
Firstly, I'm surprized to hear your network has 1000x higher permeability than your actual material. I don't generally expect PNMs to be predictive without some fiddling, but +/- 2-3x is the normal range, not 1000x.  This makes me thing something is wrong somewhere in your workflow.  Perhaps you put the wrong voxel size somewhere, for instance. 

As for your actual question, without seeing your code this is just a guess, but it seems like you need to run  the ``regenerate_models`` function on the the geometry and physics objects.  If you overwrite some values in the dependency chain you need to tell OpenPNM to update everything.

**Reply by elstavro** [reply]
Thanks again for this reply, I think I found my mistake that gives so different permeability:
The binary porosity image I'm imported has pores = 255 and solid matrix = 255, whereas it should be the inverse, i.e. pores = 0 and solid = 255.
Could you please confirm this?
Thank you!

**Comment by elstavro** [comment]
Thanks for the quick reply.
-- Experiments vs PNM: 
I believe the problem is related to the connected porosity (in reality) vs the generated connected network (snow2). For instance the permeability of the solid matrix is very low and this is what I'm trying to model by imposing very small throat diameters (which are computed as the half of the smaller connecting pore). 
This solid matrix permeability corresponds to micro-pores that I can't see with the imaging resolution I have (x-rays, 68 microns/px). 
Does this make sense? Feedback is very welcome.

-- code:  
oh great, `regenerate_models` seems to work beautifully, thanks! 
I will double check the units of dimensions.

**Comment by elstavro** [comment]
I have one more question regarding the assigned throat diameters (initially, before the modification I'm suggesting above).

I print (in m):
min pore.diameter 0.000272
max pore.diameter 0.003618724548339844
min throat.diameter 0.000272
max throat.diameter 0.003309022918701172

This is a fully connected network (I have trimmed isolated pores, checked network health and confirmed `is_fully_connected`).
If d_throat = 1/2 d_p1 (with d_p1 < d_p2) how is it possible that:
1. the min pore and trhoat diameters are the same ??
2. max d_trhoat >1/2 d_max_pore ??

Thanks in advance.
